src/app/GitUI/Editor/Diff/GrepHighlightService.cs

```diff
@@ -5,29 +5,30 @@
 using GitExtensions.Extensibility;
 using GitExtensions.Extensibility.Git;
 using GitExtUtils;
-using GitExtUtils.GitUI.Theming;
-using ICSharpCode.TextEditor;
 using ICSharpCode.TextEditor.Document;
 
 namespace GitUI.Editor.Diff;
 
 public partial class GrepHighlightService : TextHighlightService
 {
     private readonly List<TextMarker> _textMarkers = [];
-    private DiffLinesInfo _matchInfos = new();
+    private DiffLinesInfo _diffLinesInfo = new();
 
     [GeneratedRegex(@"^(?<line>\d+)(?<kind>:|.)(?<text>.*)$", RegexOptions.ExplicitCapture)]
     private static partial Regex GrepLineRegex();
 
-    public GrepHighlightService(ref string text)
-        => SetText(ref text);
+    public GrepHighlightService(ref string text, DiffViewerLineNumberControl lineNumbersControl)
+    {
+        SetText(ref text);
+        lineNumbersControl.DisplayLineNum(_diffLinesInfo, showLeftColumn: false);
+    }
+
+    public override void AddTextHighlighting(IDocument document)
+        => document.MarkerStrategy.AddMarkers(_textMarkers);
```

Comment:
❔ Move to a base class?

Reply:
That would be TextHighlightService, but there are no textMerkers there.
So overridden in Diff (base), Grep and Difftastic.